crypto/keyring/info.go

```diff
@@ -237,13 +238,30 @@ func (i multiInfo) GetPath() (*hd.BIP44Params, error) {
 	return nil, fmt.Errorf("BIP44 Paths are not available for this type")
 }
 
+// UnpackInterfaces implements UnpackInterfacesMessage.UnpackInterfaces
+func (i multiInfo) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
+	multiPK := i.PubKey.(*multisig.LegacyAminoPubKey)
+
+	return codectypes.UnpackInterfaces(multiPK, unpacker)
+}
+
 // encoding info
 func marshalInfo(i Info) []byte {
 	return CryptoCdc.MustMarshalBinaryLengthPrefixed(i)
 }
 
 // decoding info
 func unmarshalInfo(bz []byte) (info Info, err error) {
+	// We first try to unmarshal into a multiInfo. This allows the unmarshal
+	// functions to unmarshal Anys correctly too.
+	var multi multiInfo
+	err = CryptoCdc.UnmarshalBinaryLengthPrefixed(bz, &multi)
+	if err == nil {
+		return multi, nil
+	}
+
+	// The above might fail, e.g. when Info is a local, offline or ledger info.
+	// In this case, we just unmarshal into the interface.
 	err = CryptoCdc.UnmarshalBinaryLengthPrefixed(bz, &info)
 	return
```

Comment:
I'm not 100% confident about this code.

Before, we just called `CryptoCdc.UnmarshalBinaryLengthPrefixed(bz, &info)`, where `info` is an `Info` interface.

Now, because we use Any, we need to UnpackInterfaces. If we kept the old code `CryptoCdc.UnmarshalBinaryLengthPrefixed(bz, &info)`, then since the `Info` interface doesn't implement UnpackInterfaces, we'll never unpack, and the any cachedValue will be nil.

Out of the 4 concrete types `{multi,offline,ledger,local}Info`, only `multiInfo` needs to UnpackInterfaces. That's why here, what I did, is:
- first, try `CryptoCdc.UnmarshalBinaryLengthPrefixed(bz, &multi)`. Since `multiInfo` implements UnpackInterfaces, this will correctly unpack the anys in the multiInfo case.
- if the `info` was actually not a `multiInfo` but another `Info`, then the above line will error, in which case we call the old `CryptoCdc.UnmarshalBinaryLengthPrefixed(bz, &info)`.

Reply:
@amaurymartiny looks like this was the original PR. Can you push https://github.com/cosmos/cosmos-sdk/pull/7276/commits/3405510e2ecf884bc323d635c2f8594deb15473d here instead?